core/src/main/java/org/opensearch/sql/analysis/JsonSupportAnalyzer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.analysis;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.Node;
+import org.opensearch.sql.ast.expression.Alias;
+import org.opensearch.sql.ast.expression.Cast;
+import org.opensearch.sql.ast.expression.Function;
+import org.opensearch.sql.ast.expression.Literal;
+import org.opensearch.sql.ast.tree.Aggregation;
+import org.opensearch.sql.ast.tree.Filter;
+import org.opensearch.sql.ast.tree.Project;
+
+public class JsonSupportAnalyzer extends AbstractNodeVisitor<Boolean, Object> {
```

Comment:
`JsonSupportNodeVisitor`?  I'm not sure why we call things "analyzers" when they're node visitors.

Reply:
`<Boolean, Object>` doesn't seem correct, but I suppose we don't really have a ned for a context yet, nor a return type (which is usually a Plan or OptimizedTree.  
If the purpose of the node visitor is just to find unsupported nodes, then I suppose this is fine but we should include a javadoc comment that details this.

Reply:
Longer term, we may consider using this to return a JSON object or a tree that can be turned into a JSON object. 
We can remove the Boolean return type if we throw Exceptions in the node visitors. 
If not, we should be returning Boolean types, not boolean types.  Example, `Boolean.TRUE`.

Reply:
Thanks. Changed this in 74f88ae574b06eb63778a511830e0c148cda8539 and 6785f2358a23b798aff79b7e7a623c73d1fe041b. 
I've made each unsupported node visitor throw an exception so there is no more use for return type boolean. However The abstract class requires a return type so I left it as Boolean.